### PR TITLE
Add Textfield component

### PR DESCRIPTION
**Pull request by MehriGolchin**

### CODE REVIEW

File: src/components/textfield/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    readonly title: string;
+    readonly placeholder?: string;
+    readonly name?: string;
+    readonly value?: string;
+}
+
+export class TextField extends Component<TextFieldProps> {
+    constructor(props: TextFieldProps) {
+        super(props);
+        this.state = {};
```
Comment (MehdiGolchin) [inline src/components/textfield/index.tsx]: Why did you set the state, although you did not use it anywhere in your code?

File: src/components/textfield/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    render() {
+        const title = this.props.title;
+        const placeholder = this.props.placeholder;
+        const value = this.props.value;
```
Comment (MehdiGolchin) [inline src/components/textfield/index.tsx]: You can use destructuring assignment here.
```javascript
const  { title, placeholder, value } = this.props;
```

File: src/components/textfield/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return (
+            <div>
+                <Label>{title}
```
Comment (MehdiGolchin) [inline src/components/textfield/index.tsx]: AFAIK, we do not put an input element within a label tag. Please take a look at [Mozilla doc](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label). ✋

File: src/components/textfield/textfield.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { TextField } from './index';
+
+storiesOf('TextField', module)
+  .add('default', () => (
+    <TextField title="CHOOSE A USER NAME" />
```
Comment (MehdiGolchin) [inline src/components/textfield/textfield.stories.tsx]: Why did you use "Choose a user name" for the demo? 🤔

File: src/components/textfield/textfield.styles.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import styled, { css } from 'styled-components';
```
Comment (MehdiGolchin) [inline src/components/textfield/textfield.styles.ts]: You did not use `css` anywhere in your code. 😉

File: src/components/textfield/textfield.styles.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+export const Label = styled.label`
+  color: #696969;
```
Comment (MehdiGolchin) [inline src/components/textfield/textfield.styles.ts]: It's better to use a theme here. Please do not use explicit values for font-families, font-size, colors, ... 💄

**Comment by MehdiGolchin** [comment]
Please do not forget to choose a meaningful title for your PR. I mean something that shows your What. Plus, write down a short description of your Why and How.😉

File: src/components/textfield/index.tsx (hunk 1 above)
Comment (MehriGolchin) [reply src/components/textfield/index.tsx]: I don't know :))))

File: src/components/textfield/index.tsx (hunk 3 above)
Comment (MehriGolchin) [reply src/components/textfield/index.tsx]: But it mentions that we should use this way if we don't have "id".

File: src/components/textfield/index.tsx (hunk 2 above)
Comment (MehriGolchin) [reply src/components/textfield/index.tsx]: Done!

File: src/components/textfield/textfield.stories.tsx (hunk 4 above)
Comment (MehriGolchin) [reply src/components/textfield/textfield.stories.tsx]: I put "Label". if you want to change, say me.

File: src/components/textfield/textfield.styles.ts (hunk 5 above)
Comment (MehriGolchin) [reply src/components/textfield/textfield.styles.ts]: Removed it.

File: src/components/textfield/textfield.styles.ts [hunk 7]
```diff
@@ -1,16 +1,14 @@
...
 export const Input = styled.input`
-  display: block;
-  line-height: 30px;
-  width: 200px;
-  font-family: Arial;
-  font-size:20px;
-  color: #696969;
-`;
\ No newline at end of file
+  display: ${(props: any) => props.theme.inputDisplay};
```
Comment (MehdiGolchin) [inline src/components/textfield/textfield.styles.ts]: You shouldn't put everything in the theme. 😆

File: src/components/textfield/textfield.styles.ts [hunk 8]
```diff
@@ -1,16 +1,14 @@
...
+  line-height: ${(props: any) => props.theme.inputLineheight};
+  font-family: ${(props: any) => props.theme.fontfamily};
+  Width: ${(props: any) => props.theme.inputWidth};
```
Comment (MehdiGolchin) [inline src/components/textfield/textfield.styles.ts]: Please don't use explicit `width`, we should think about responsive design. 😉

Comment (MehriGolchin) [reply src/components/textfield/textfield.styles.ts]: Done!

File: src/components/textfield/textfield.styles.ts (hunk 7 above)
Comment (MehriGolchin) [reply src/components/textfield/textfield.styles.ts]: Done!

**Pull request merged**